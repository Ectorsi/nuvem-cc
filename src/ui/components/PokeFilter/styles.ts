import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: space-around;

    & > h2 {
        margin-bottom: 15px;
        font-weight: bold;
    }

    & > p {
        color: black;
        font-weight: 500;
        margin-bottom: 10px;
        align-self: center;
    }
`;

export const SelectContainer = styled.div`
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: end;
    justify-content: space-around;
    flex-wrap: wrap;

    margin-bottom: 16px;
`;
