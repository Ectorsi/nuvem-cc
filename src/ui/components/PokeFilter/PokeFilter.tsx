import React from "react";
import * as S from "./styles";
import InputText, { InputTextProps } from "../InputText/InputText";
import SelectBox, { SelectProps } from "../SelectBox/SelectBox";
import Button from "../Button/Button";


export type PokeFilterProps = {
    inputTextProps: InputTextProps;
    selectBoxRegion: SelectProps;
    selectBoxLocation: SelectProps;
    selectBoxAreas: SelectProps;
    cleanFilters: () => void;
};

const PokeFilter = ({ inputTextProps, selectBoxRegion, selectBoxLocation, selectBoxAreas, cleanFilters }: PokeFilterProps) => {

    return (
        <S.Container>
            <h2>Pesquise o Pokemon pelo nome</h2>
            <InputText {...inputTextProps} />
            <h2>Filtrar pokemons por região/área</h2>
            <p>Para filtrar um pokemon por regiao/area, você deve preencher todos os filtros!</p>
            <S.SelectContainer>
                <SelectBox {...selectBoxRegion} />
                <SelectBox {...selectBoxLocation} />
                <SelectBox {...selectBoxAreas} />
                <Button onClick={cleanFilters}>Limpar filtros</Button>
            </S.SelectContainer>
        </S.Container>
    )
};

export default PokeFilter;