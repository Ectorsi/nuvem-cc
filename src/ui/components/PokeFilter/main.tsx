import React, { useState, MouseEvent } from "react";
import * as S from "./styles";
import Select from "../../components/Select/main";

type PokeFilterProps = { // TODO verificar quais tipos posso usar
    name?: string;
    rest?: any;
};

const PokeFilter = ({ name, ...rest }: PokeFilterProps) => {
    const [isOpen, setIsOpen] = useState(false);

    const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
        if (e) {
            setIsOpen(!isOpen);
        }
    };

    return (
        <S.Container>
            {/* <Select placeholder="Filtre por regiÃ£o">
                <option value="1">1</option>
                <option value="2">2</option>
                <option value="3">3</option>
            </Select> */}
        </S.Container>
    )
};

export default PokeFilter;