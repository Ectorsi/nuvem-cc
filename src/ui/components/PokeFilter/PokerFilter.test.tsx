import { render, screen } from '@testing-library/react';

import PokeFilter, { PokeFilterProps } from './PokeFilter';

const mockPokeFilter: PokeFilterProps = {
    clearSearch: jest.fn(),
    cleanFilters: jest.fn(),
    inputTextProps: {
        placeholder: 'Test',
        value: 'Test',
        onChange: jest.fn(),
        error: '',
    },
    selectBoxRegion: {
        label: 'LabelTest',
        placeholder: 'PlaceholderTest',
        value: 'SelectrValue',
        onChange: jest.fn(),
        error: '',
        options: [
            { value: 'Option1', label: 'Option1' },
            { value: 'Option2', label: 'Option2' },
        ],
    },
    selectBoxLocation: {
        label: 'LabelTest',
        placeholder: 'PlaceholderTest',
        value: 'SelectrValue',
        onChange: jest.fn(),
        error: '',
        options: [
            { value: 'Option1', label: 'Option1' },
            { value: 'Option2', label: 'Option2' },
        ],
    },
    selectBoxAreas: {
        label: 'LabelTest',
        placeholder: 'PlaceholderTest',
        value: 'SelectrValue',
        onChange: jest.fn(),
        error: '',
        options: [
            { value: 'Option1', label: 'Option1' },
            { value: 'Option2', label: 'Option2' },
        ],
    },
};

describe('PokeFilter unit test', () => {
    it('should be able to match snapshot', () => {
        const { container } = render(<PokeFilter {...mockPokeFilter} />);
        expect(container.firstChild).toMatchSnapshot();
    });
    it('should be able to clear fields', () => {
        render(<PokeFilter {...mockPokeFilter} />);
        const button = screen.getByText('Limpar filtros');
        button.click();
        expect(mockPokeFilter.clearSearch).toHaveBeenCalled();
        expect(mockPokeFilter.cleanFilters).toHaveBeenCalled();
    });
});
