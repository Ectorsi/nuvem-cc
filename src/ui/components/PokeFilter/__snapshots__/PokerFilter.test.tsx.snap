// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeFilter unit test should be able to match snapshot 1`] = `
<div
  class="sc-beqWaB hiqCdz"
>
  <h2>
    Pesquise o Pokemon pelo nome
  </h2>
  <input
    class="sc-dmqHEX hlUAwi"
    placeholder="Test"
    type="text"
    value="Test"
  />
  <h2>
    Filtrar pokemons por região/área
  </h2>
  <p>
    Para filtrar um pokemon por regiao/area, você deve preencher todos os filtros!
  </p>
  <div
    class="sc-gueYoa ienTpA"
  >
    <div
      class="sc-iAEyYk jEYiaI"
    >
      <label
        class="sc-qRumB RfdGC"
      >
        LabelTest
      </label>
      <select
        class="sc-fsQiph NiIQK"
      >
        <option
          value=""
        >
          PlaceholderTest
        </option>
        <option
          value="Option1"
        >
          Option1
        </option>
        <option
          value="Option2"
        >
          Option2
        </option>
      </select>
    </div>
    <div
      class="sc-iAEyYk jEYiaI"
    >
      <label
        class="sc-qRumB RfdGC"
      >
        LabelTest
      </label>
      <select
        class="sc-fsQiph NiIQK"
      >
        <option
          value=""
        >
          PlaceholderTest
        </option>
        <option
          value="Option1"
        >
          Option1
        </option>
        <option
          value="Option2"
        >
          Option2
        </option>
      </select>
    </div>
    <div
      class="sc-iAEyYk jEYiaI"
    >
      <label
        class="sc-qRumB RfdGC"
      >
        LabelTest
      </label>
      <select
        class="sc-fsQiph NiIQK"
      >
        <option
          value=""
        >
          PlaceholderTest
        </option>
        <option
          value="Option1"
        >
          Option1
        </option>
        <option
          value="Option2"
        >
          Option2
        </option>
      </select>
    </div>
    <button
      class="sc-kFuwaP hSKZFG"
    >
      Limpar filtros
    </button>
  </div>
</div>
`;
